timemachine/cpp/src/barostat.cu

```diff
@@ -260,57 +390,45 @@ void MonteCarloBarostat::inplace_move(
             nullptr,
             nullptr,
             nullptr,
-            d_u_after_,
-            stream // TBD: parallelize me!
+            d_u_after_buffer_,
+            stream
         );
     }
 
+    cub::DeviceReduce::Sum(
+        d_sum_storage_,
+        d_sum_storage_bytes_,
+        d_u_after_buffer_,
+        d_final_u_,
+        N_,
+        stream
+    );
+    gpuErrchk(cudaPeekAtLastError());
+
     double pressure = pressure_*AVOGADRO*1e-25;
     const double kT = BOLTZ*temperature_;
 
-    unsigned long long u_init_agg = 0;
-    unsigned long long u_final_agg = 0;
-
-    unsigned long long initial_energy[N_];
-    unsigned long long final_energy[N_];
-
-    gpuErrchk(cudaMemcpyAsync(initial_energy, d_u_before_, N_*sizeof(*d_u_before_), cudaMemcpyDeviceToHost, stream));
-    gpuErrchk(cudaMemcpyAsync(final_energy, d_u_after_, N_*sizeof(*d_u_after_), cudaMemcpyDeviceToHost, stream));
-
-    gpuErrchk(cudaStreamSynchronize(stream));
-
-    for (int i = 0; i < N_; i++) {
-        u_init_agg += initial_energy[i];
-        u_final_agg += final_energy[i];
-    }
-
-    double u_init = FIXED_TO_FLOAT<double>(u_init_agg);
-    double u_final = FIXED_TO_FLOAT<double>(u_final_agg);
-
-    double w = u_final-u_init + pressure*delta_volume - num_molecules*kT*std::log(new_volume/volume);
+    const int move_blocks = (N_ + tpb - 1) / tpb;
```

Comment:
Could you comment on intent of this line? (`ceil(N_ / tpb)`?)

Reply:
Correct, this is the integer ceil pattern in C++.